extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfig.java

```diff
@@ -0,0 +1,129 @@
+package io.quarkus.mailer.impl;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "mailer", phase = ConfigPhase.RUN_TIME)
+public class MailConfig {
+
+    /**
+     * Configure the default `from` attribute.
+     * It's the sender email address.
+     */
+    @ConfigItem
+    public Optional<String> from;
+
+    /**
+     * Enables the mock mode, not sending emails.
+     * The content of the emails is printed on the console.
+     */
+    @ConfigItem
+    public Optional<Boolean> mock;
+
+    /**
+     * Configures the default bounce email address.
+     */
+    @ConfigItem
+    public Optional<String> bounceAddress;
+
+    /**
+     * The SMTP host name.
+     */
+    @ConfigItem(defaultValue = "localhost")
+    public String host;
+
+    /**
+     * The SMTP port.
+     */
+    @ConfigItem
+    public OptionalInt port;
+
+    /**
+     * The username.
+     */
+    @ConfigItem
+    public Optional<String> username;
+
+    /**
+     * The password.
+     */
+    @ConfigItem
+    public Optional<String> password;
```

Comment:
We need to do something about clear passwords in config.

Reply:
Like what?

Reply:
In WildFly ((cue groaning)) we have a concept of a "credential store" that can be referenced from the configuration.  The credential store would contain all the sensitive stuff and could be protected using various means.  In the Quarkus case we'd probably want to delegate to OpenShift or similar to hold security credentials whenever possible.

In fact we even go a bit farther in WildFly by having a complete separate configuration governing client authentication, which matches outbound requests by protocol, host or IP address, abstract type, etc. and specifies the type of authentication to perform (which might include e.g. OAUTH or Kerberos or other non-user-password strategy) along with all of the associated parameters.  This might or might not be pushing things a bit too far for our use case - but it might be worth discussing anyway.

Reply:
In my experience, if the application can extract the sensitive config by itself, it's only ever just slightly obfuscated, so putting the passwords in plain text does not really hurt.

If outside people have access to the deployment, however, such as by being able to view the source code or browse config files, then the best way to pass those passwords remains like OpenShift does and pass them as env vars from a source that is protected.

So in both cases I'm not sure if we need to do something special to hide passwords. I believe we already support passing config values via env vars.

Reply:
Right, the most common problem has less to do with deployment and more to do with e.g. the fact that our issue template asks the user to paste their `application.properties`. ;-)  Many users do not consider configuration files to be sensitive and will happily email them, paste them to the web, etc.  It's better to at least put passwords into a separate file (which can then be `chmod 0400` for example) to help avoid accidents.

Reply:
OK, but in that case let's FINALLY support config files like this:

In `application.properties`
```
quarkus.smtp.host=myhost
# I don't care about the syntax
@include passwords.properties
```

In `passwords.properties`:
```
quarkus.smtp.password=secret
```

Or, you know, make it a convention, and OOTB load `passwords.properties` if it exists.

Reply:
I've opened https://github.com/quarkusio/quarkus/issues/2648 to discuss it.